### PR TITLE
Prescriptions init v2

### PR DESCRIPTION
**Pull request by zcox10**
## Description
Create a Prescriptions skeleton with the basic schema for the data model and some initial functions.  Additional basic cleanup defined below:
- Create initial Prescription structure similar to the User model
- Removed `.idea` artifact from original commit in repo with `git rm -r --cached .idea`
- Added a `pull_request_template.md` to `.github`

Changes from the previous PR:
- Added foreign key annotations `@JoinColumn` to `medication_id` and `user_id`
- Added `num_doses` and `is_active` fields to `prescriptions`
- Removed getters/setters from Prescription.java and added a `@Data` annotation to the Prescription class
- Clarification that I'm removing `.idea` from the main branch, not committing any `.idea` artifacts

## Checklist

- [X] Tests pass
- [X] No checkstyle errors
- [X] No PMD errors
- [X] No compilation/build errors

### CODE REVIEW

File: src/main/java/com/bytecoders/pharmaid/repository/PrescriptionRepository.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import com.bytecoders.pharmaid.repository.model.Prescription;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+/**
+ * JPA prescriptions repository.
+ */
+public interface PrescriptionRepository extends JpaRepository<Prescription, String> {
+
+  Optional<Prescription> findByUserId(String userId);
```
Comment (AlexLoyko) [inline src/main/java/com/bytecoders/pharmaid/repository/PrescriptionRepository.java:12]: correct to assume this is for future use?

Comment (zcox10) [reply src/main/java/com/bytecoders/pharmaid/repository/PrescriptionRepository.java:12]: Yes, thinking was that if we need to find all of a user's prescriptions, we would find the user first, then get their prescriptions.  Based it off of what I was reading with the `findByEmail()` function for `loginUser()`. 

Let me know if that is the right approach

Comment (AlexLoyko) [reply src/main/java/com/bytecoders/pharmaid/repository/PrescriptionRepository.java:12]: looks great

**Comment by AlexLoyko** [comment]
Which parts did you tests e2e with the DB?

**Pull request merged**